### DISCUSSION TITLE
Is Iroh censorship resistant?

### DISCUSSION THREAD

**Discussion by markxueyuan** [n0-computer/iroh#1924] [General]
Hi!

I played with Iroh for a whole day yesterday! It was a real fun!

I am from China. Currently Iroh runs perfectly well in China.

But in the long term, the Chinese government could probably block Iroh, as it did to many communicating tools, such as telegram, discord, twitter, facebook, etc.

I wonder as a decentralized web3 protocol, whether Iroh has some mechanism to realize censorship resistance?

**Comment by rklaehn** [comment]
Hi. Glad that you like it.

This question is mostly about iroh-net.

Iroh net has two modes, it is able to establish direct connections and use DERP relays to do NAT hole punching and to relay data as a fallback.

The traffic flowing over the DERP relay is fully encrypted, and the relay itself does not know more than what is necessary to do it's job.

If iroh becomes successful, it is possible that a hostile government will block the n0 relay network. But we have recently made a change so that iroh tickets contain derp URLs instead of derp IDs, so projects that care very much about privacy and censorship resistance can run their own relays. This was done due to customer feedback from such projects that care about privacy.

So TLDR: is the iroh default config going to work out of the box once we become sufficiently successful to appear on their radar? Probably not. Can you still use iroh and tools build on iroh-net such as like sendme, dumbpipe etc? Very likely yes.
